### PR TITLE
port to new ray train api in estimators

### PR DESCRIPTION
**Pull request by kira-lin**
Currently used ray.train.Trainer has been deprecated. In addition, the get_model method is not working now. This PR updates to use TorchTrainer and TensorflowTrainer and implements get_model based on checkpoint.

### CODE REVIEW

File: python/raydp/torch/estimator.py [hunk 1]
```diff
@@ -211,15 +215,15 @@ def train_func(config):
         metrics = config["metrics"]
 
         # create dataset
-        train_data_shard = get_dataset_shard("train")
+        train_data_shard = session.get_dataset_shard("train")
```
Comment (carsonwang) [inline python/raydp/torch/estimator.py:218]: The latest Ray document suggests using `iter_torch_batches` instead of `to_torch` on the data shard. Can you please take a look at the difference?

**Comment by carsonwang** [comment]
@KepingYan , can you please take a look as well?

**Review by KepingYan — COMMENTED** [review]
LGTM.

File: python/raydp/torch/estimator.py [hunk 2]
```diff
@@ -334,17 +353,14 @@ def fit_on_spark(self,
...
-
-    def restore(self, checkpoint):
-        assert self._trainer is not None, "Must call fit first"
-        self._trainer.load(checkpoint)
-
-    def shutdown(self):
-        if self._trainer is not None:
-            self._trainer.shutdown()
-            self._trainer = None
+        states = self._trained_results.checkpoint.to_dict()["state_dict"]
```
Comment (carsonwang) [inline python/raydp/torch/estimator.py:356]: Why not calling `checkpoint.get_model()` to get the model?

File: python/raydp/torch/estimator.py [hunk 3]
```diff
@@ -233,12 +235,20 @@ def train_func(config):
...
                                                                             model, loss, metrics)
-                train.report(epoch=epoch, evaluate_res=evaluate_res, test_loss=evaluate_loss)
+                session.report(dict(epoch=epoch, eval_res=eval_res, test_loss=evaluate_loss))
                 loss_results.append(evaluate_loss)
+        if hasattr(model, "module"):
+            states = model.module.state_dict()
+        else:
+            # if num_workers = 1, model is not wrapped
+            states = model.state_dict()
+        session.report({}, checkpoint=Checkpoint.from_dict({
```
Comment (carsonwang) [inline python/raydp/torch/estimator.py:249]: This saves the checkpoint for every epoch. Is this a default behavior? Otherwise we also don't need to do it by default.

File: python/raydp/torch/estimator.py (hunk 2 above)
Comment (kira-lin) [reply python/raydp/torch/estimator.py:356]: To use get_model, we need to use framework specific checkpoint, namely TorchCheckpoint and TensorflowCheckpoint, these are in "public api beta" phase. They do the same thing as we do here.

File: python/raydp/torch/estimator.py (hunk 3 above)
Comment (kira-lin) [reply python/raydp/torch/estimator.py:249]: I tested removing this, but the checkpoint included in the returned result does not contain anything.

File: python/raydp/tf/estimator.py [hunk 4]
```diff
@@ -45,13 +49,16 @@ def __init__(self,
                  label_shape: Optional[tf.TensorShape] = None,
                  batch_size: int = 128,
                  num_epochs: int = 1,
-                 shuffle: bool = True,
+                 shuffle: bool = False,
```
[the remark this answers is not in the document]
Comment (kira-lin) [reply python/raydp/tf/estimator.py:52]: Do we want to make this default? this might hurt performance for users who don't need it

[the remark this answers is not in the document]
Comment (carsonwang) [reply python/raydp/tf/estimator.py:52]: Yes, let's keep the default as "True". It is common to do a global shuffle before training. Other frameworks like horovod estimator set the default to true as well.

File: python/raydp/torch/estimator.py (hunk 3 above)
Comment (carsonwang) [reply python/raydp/torch/estimator.py:249]: Where is checkpoint stored? Are anything in the path?

File: python/raydp/torch/estimator.py (hunk 2 above)
Comment (carsonwang) [reply python/raydp/torch/estimator.py:356]: I see. Let's use our own code for now.

File: python/raydp/tf/estimator.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kira-lin) [reply python/raydp/tf/estimator.py:52]: ok

File: python/raydp/torch/estimator.py (hunk 3 above)
Comment (kira-lin) [reply python/raydp/torch/estimator.py:249]: It is stored in ~/ray_results, there are some pickled binary files. But when I use TorchCheckpoint.from_directory to load it, get_model does not work.
